src/ol/render/webgl/webglreplay.js

```diff
@@ -1433,46 +1428,21 @@ ol.render.webgl.LineStringReplay.prototype.drawReplay_ = function(gl, context, s
     end = this.startIndices_[this.startIndices_.length - 1];
     this.drawElements_(gl, context, 0, end);
   }
-  gl.clear(gl.DEPTH_BUFFER_BIT);
-  gl.disable(gl.DEPTH_TEST);
-  //Restore GL parameters.
-  //NOTE: Type checks will be obsolete when Closure Compiler recognizes the return type of specific
-  //gl.getParameter() calls.
-  if (typeof tmpDepthMask === 'boolean') {
-    gl.depthMask(tmpDepthMask);
-  }
-  if (typeof tmpDepthFunc === 'number') {
-    gl.depthFunc(tmpDepthFunc);
+  if (!hitDetection) {
+    gl.clear(gl.DEPTH_BUFFER_BIT);
+    gl.disable(gl.DEPTH_TEST);
+    //Restore GL parameters.
+    //NOTE: Type checks will be obsolete when Closure Compiler recognizes the return type of specific
+    //gl.getParameter() calls.
+    if (typeof tmpDepthMask === 'boolean') {
+      gl.depthMask(tmpDepthMask);
```

Comment:
You can use `var tmpDepthMask = /** @type {boolean} */ (gl.getParameter(gl.DEPTH_FUNC));` to tell the compiler the type.